components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -413,6 +420,22 @@ - (void)setSelectionIndicatorTemplate:
   }
 }
 
+- (void)setContentPadding:(UIEdgeInsets)contentPadding
+           forLayoutStyle:(MDCTabBarViewLayoutStyle)layoutStyle {
```

Comment:
suggestion: You could optimize this by checking if the contentPadding is already the same as the one stored in `layoutStyleToContentPadding` and doing an early return.

This could be particularly useful to avoid unnecessary `setNeedsLayout` if it hasn't really changed.

Reply:
I've considered it, but I'd rather not introduce an optimization that could lead to bugs until we have some indication that it's a performance problem. I would expect this API to only be called immediately after initialization. Even if it's called multiple times in a row, the dirty layout flag should only take effect once on the next layout pass. 

Is there a situation where you think clients might be calling this more frequently or at regular intervals?

Reply:
I think this is a pretty common iOS practice to optimize redundant calls to setters.

But I also don't expect this value to change much, so in this case it probably isn't that critical.

Reply:
Thanks. MDC-iOS has experienced problems in the past where optimizations like that didn't produce the expected side-effects and subtle bugs arose.  I'll leave the unoptimized code until we can identify that it's causing performance issues.